analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -450,6 +463,425 @@ public void Finally_NestedFinally()
                 "InInnerFinally",
                 "AfterInnerFinally",
                 "AfterOuterFinally");
+
+            validator.TagStates("InOuterFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerTry").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+
+            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
```

Comment:
Did you check that this has the same cfg as `catch(Exception)` case?

Reply:
Ha. Interesting. The CGF is the same but the behavior is different.